### PR TITLE
Created a github action file

### LINKED ISSUE #70: Creating a prettier Github action
image

##  Welcome to NodeCloud with [Hacktoberfest 2020](https://hacktoberfest.digitalocean.com/)! 
**Stay in the know:** Get notified of updates by clicking the 👁‍🗨 watch button and save the repository for later reference by clicking the ⭐️ star button at the top right corner of this repo! 

### Creating a prettier Github action 

Prettier is a very popular code formatter to format your code and prevent the ongoing debates about code styles.  Currently, prettier is triggered through a pre-commit hook ([husky](https://typicode.github.io/husky/#/))  which will run Prettier on the staged files prior to committing.  

This works well in most cases, but there are certain cases where the code might not be formatted:

- Running `git commit --no-verify` will commit code without running pre-commit hooks which can result in files not being formatted.

- Pre-commit hook ([husky](https://typicode.github.io/husky/#/)) is not properly installed in the developers local machine.

To automate this process you can use  [git auto-commit action ](https://github.com/stefanzweifel/git-auto-commit-action) to run prettier and commit the changes if necessary. Check out the references to get started 🚀! 

 
### Additional information

* The project currently contains two main branches: `master` and `nodecloud-next`. **This feature is required to the `nodecloud-next` branch. so make sure that you create the feature branch from `nodecloud-next`.** 

**`nodecloud-next:`** https://github.com/leopardslab/nodecloud/tree/nodecloud-next

* Be sure to sign up at https://hacktoberfest.digitalocean.com so you can qualify to pick a limited edition t-shirt 👚 or plant a tree 🌳!

* See https://hacktoberfest.digitalocean.com for more details

### Impact

*  Github workflows directory 

### References

* Helping [article](https://medium.com/@mskelton/auto-formatting-code-using-prettier-and-github-actions-ed458f58b7df) and [sample project ](https://github.com/mskelton/prettier-formatting-example) by @mskelton

* [git auto-commit action ](https://github.com/stefanzweifel/git-auto-commit-action) by @stefanzweifel

### PR DESCRIPTION
**Pull request by lumenCodes**
Signed-off-by: cld <lumenluwiz@gmail.com>

Fixes #70 

## Proposed Changes

  - This is a code formatter Github action file

### CODE REVIEW

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+      - uses: actions/setup-node@v1
+        with:
+          node-version: "12.x"
+      - name: Format
+        run: |
+          npm ci
+          npm run format
```
Comment (rajitha1998) [inline .github/workflows/format.yml]: Since `nodecloud` uses yarn instead of the npm client, this needs to be replaced with,

```
yarn
yarn format
```

Comment (rajitha1998) [reply .github/workflows/format.yml]: Also, the format command needs to be updated in the `package.json` as mentioned in the article. I will look more into that and let you know.

Comment (lumenCodes) [reply .github/workflows/format.yml]: Ok.

Comment (rajitha1998) [reply .github/workflows/format.yml]: you need to add these line in the `package.json` file

"format": "prettier --write ."

The rest LGTM  :) @lumenCodes

Comment (lumenCodes) [reply .github/workflows/format.yml:18]: As a dev dependency?

Comment (lumenCodes) [reply .github/workflows/format.yml]: Done @rajitha1998

File: package.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
   ],
   "author": "scorelab",
   "license": "Apache-2.0",
+  "format": "prettier --write ."
```
Comment (rajitha1998) [inline package.json]: Hey @lumenCodes, the executing scripts will be added to the `scripts` section of the `package.json` file.
```
"scripts": {
    "test": "lerna run test",
    "lint": "lerna run lint",
    "prettier": "lerna run prettier",
    "doc": "jsdoc -c jsdoc.json"
  },
```

**Comment by rajitha1998** [comment]
Great work @lumenCodes!  Hope you learnt something new from the Node Cloud project in Hacktoberfest!

This pull request LGTM, it will be merged once it is reviewed by a senior member.  Thank you so much for your keen interest to the [SCoRe Lab community](https://gitter.im/scorelab/scorelab).

**Review by rajitha1998 — APPROVED** [review]

**Comment by lumenCodes** [comment]
Thanks so much for your help I really appreciate 😊 . And yes I learnt quite a few things, I hope to keep contributing to the project.

**Pull request merged**